src/aspire/utils/blk_diag_matrix.py

```diff
@@ -855,22 +854,28 @@ def filter_to_fb_mat(h_fun, fbasis):
 
     # Represent 1D function values in fbasis
     h_fb = BlkDiagMatrix.empty(2 * fbasis.ell_max + 1, dtype=fbasis.dtype)
-    ind = 0
+    ind_ell = 0
+    ind_radial = 0
     for ell in range(0, fbasis.ell_max+1):
         k_max = fbasis.k_max[ell]
         rmat = 2*k_vals.reshape(n_k, 1)*fbasis.r0[0:k_max, ell].T
         fb_vals = np.zeros_like(rmat)
         for ik in range(0, k_max):
-            fb_vals[:, ik] = jv(ell, rmat[:, ik])
-        fb_nrms = 1/np.sqrt(2)*abs(jv(ell+1, fbasis.r0[0:k_max, ell].T))/2
-        fb_vals = fb_vals/fb_nrms
+            # filters use different normalized factors
+            if ell == 0:
```

Comment:
The only difference between the two branches is the factor `np.sqrt(2)`. Move the copy from `fbasis._precomp` out and have a single `if ell == 0` statement that multiplies by `np.sqrt(2)`.

Reply:
Done. The inner loop was also removed in commit 01c8fcd.